linera-rpc/src/grpc/client.rs

```diff
@@ -434,6 +437,26 @@ impl ValidatorNode for GrpcClient {
         Ok(certs_collected)
     }
 
+    #[instrument(target = "grpc_client", skip(self), err(level = Level::WARN), fields(address = self.address))]
+    async fn download_certificates_by_heights(
+        &self,
+        chain_id: ChainId,
+        heights: Vec<BlockHeight>,
+    ) -> Result<Vec<ConfirmedBlockCertificate>, NodeError> {
+        let request = CertificatesByHeightRequest { chain_id, heights };
+        let received: Vec<ConfirmedBlockCertificate> = Vec::<Certificate>::try_from(
+            client_delegate!(self, download_certificates_by_heights, request)?,
+        )?
+        .into_iter()
+        .map(|cert| {
+            ConfirmedBlockCertificate::try_from(cert)
```

Comment:
Can't we use the `ConfirmedBlockCertificate` type directly in the protocol, so that this would be a serialization error?

Reply:
I don't understand – we use it in the protocol

Reply:
The issue here is that the response is a `Certificate` enum – either validatedo, confirmed or timeout.

Reply:
True; not in this PR at least then. :sweat_smile: